### ISSUE TITLE
Can't print RaggedTensor with dtype uint8

### ISSUE THREAD

**Issue by fredo838** [tensorflow/tensorflow#48937]
Following code block
`>>> tf.print(tf.ragged.constant([[1,2,3],[1,2]], dtype=tf.uint8)`
raises 
```
tensorflow.python.framework.errors_impl.InvalidArgumentError: Value for attr 'T' of uint8 is not in the list of allowed values: int8, int16, int32, int64, complex64, complex128, float, double, bool, variant
	; NodeDef: {{node AsString}}; Op<name=AsString; signature=input:T -> output:string; attr=T:type,allowed=[DT_INT8, DT_INT16, DT_INT32, DT_INT64, DT_COMPLEX64, DT_COMPLEX128, DT_FLOAT, DT_DOUBLE, DT_BOOL, DT_VARIANT]; attr=precision:int,default=-1; attr=scientific:bool,default=false; attr=shortest:bool,default=false; attr=width:int,default=-1; attr=fill:string,default=""> [Op:AsString]
```
I'm filing this as a bug because I really don't think this error is thrown intentionally. If it is, apologies. 

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):  yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): tested in `2.4.1` and `2.6.0-dev20210506` (latest dev I could find)
- Python version: 3.8

**Describe the current behavior**
It raises an errror
**Describe the expected behavior**
The Ragged tensor can be printed

**[Contributing](https://www.tensorflow.org/community/contribute)** - Do you
want to contribute a PR? (yes): sure, I have absolutely no idea where to start though. 

**Standalone code to reproduce the issue**
`tf.print(tf.ragged.constant([[1,2,3],[1,2]], dtype=tf.uint8) ` in an interactive python shell

**Comment by sachinprasadhs** [comment]
I was able to reproduce your issue, please find the [gist](https://colab.research.google.com/gist/sachinprasadhs/c7853b066d1be5ac774da6022a759c90/untitled3.ipynb#scrollTo=FgJEmmVPm-3B) here.
You can try with normal `print` statement since` tf.print` is having issues.

**Comment by fredo838** [comment]
Normal `print` is not really an option, as I need to print my `RaggedTensor` every time the graph runs, not only at graph construction time.

**Comment by edloper** [comment]
Thanks for pointing this out.  The underlying issue is that [`tf.strings.as_string`](https://www.tensorflow.org/api_docs/python/tf/strings/as_string) only supports a limited set of dtypes (int8, int16, int32, int64, complex64, complex128, float32, float64, bool); and tf.print uses tf.strings.as_string to print ragged tensors.

A quick solution on the user side is to just cast the tensor to one of those types before printing.  In this case, the smallest type that would give the same string outputs would be int16, so you could do:

```
x = tf.ragged.constant([[1,2,3],[1,2]], dtype=tf.uint8)
tf.print(tf.cast(x, tf.int16))
```

On the TensorFlow side, we could do any of the following:

1. Update `AsString` (`third_party/tensorflow/core/kernels/as_string_op.cc`) to properly handle more dtypes.
2. Add a wrapper in `string_ops.py` that automatically casts `input` to an appropriate type when necessary.
3. Update `ragged_tensor_to_string` to either cast values when necessary, or fall back on a less readable result (where flat_values and row_splits are listed separately) when an unsupported dtype is used.

Option 1 seems best to me -- I don't see any particular downside to having AsString support uint8.  And presumably supporting all the other types (qint8, etc.) wouldn't be too difficult.  For tf.variant and tf.resource, we would probably still raise an exception.

**Comment by edloper** [comment]
(Minor correction: it looks like AsString *does* support variant, and uses its DebugString method.  And ResourceBase also has a DebugString method, so maybe we could use that as well, and have tf.strings.as_string support all dtypes.)

**Comment by rohan100jain** [comment]
As Ed suggested, please feel free to send us a PR adding support for more dtypes for the AsString op in TF.

**Comment by ymodak** [comment]
This is fixed with latest tf-nightly version 2.6.0-dev20210618
See [gist](https://colab.research.google.com/gist/ymodak/d00a411d185b0fcac8ba7a814336c0f0/untitled3.ipynb) for your reference. Thanks!
